core/src/main/java/org/apache/iceberg/io/DeleteWriteResult.java

```diff
@@ -32,25 +32,39 @@
 public class DeleteWriteResult {
   private final List<DeleteFile> deleteFiles;
   private final CharSequenceSet referencedDataFiles;
+  private final List<DeleteFile> rewrittenDeleteFiles;
 
   public DeleteWriteResult(DeleteFile deleteFile) {
     this.deleteFiles = Collections.singletonList(deleteFile);
     this.referencedDataFiles = CharSequenceSet.empty();
+    this.rewrittenDeleteFiles = Collections.emptyList();
   }
 
   public DeleteWriteResult(DeleteFile deleteFile, CharSequenceSet referencedDataFiles) {
     this.deleteFiles = Collections.singletonList(deleteFile);
     this.referencedDataFiles = referencedDataFiles;
+    this.rewrittenDeleteFiles = Collections.emptyList();
   }
 
   public DeleteWriteResult(List<DeleteFile> deleteFiles) {
     this.deleteFiles = deleteFiles;
     this.referencedDataFiles = CharSequenceSet.empty();
+    this.rewrittenDeleteFiles = Collections.emptyList();
   }
 
   public DeleteWriteResult(List<DeleteFile> deleteFiles, CharSequenceSet referencedDataFiles) {
     this.deleteFiles = deleteFiles;
     this.referencedDataFiles = referencedDataFiles;
+    this.rewrittenDeleteFiles = Collections.emptyList();
+  }
+
+  public DeleteWriteResult(
```

Comment:
If I had to implement `DeleteWriteResult` now, I'd add a builder like in `WriteResult`. That said, we already have 4 public constructors that we will have to maintain. I am not sure adding a builder would help now.